be/src/olap/single_replica_compaction.cpp

```diff
@@ -399,8 +401,18 @@ Status SingleReplicaCompaction::_download_files(DataDir* data_dir,
     MonotonicStopWatch watch;
     watch.start();
     for (auto& file_name : file_name_list) {
-        auto remote_file_url = remote_url_prefix + file_name;
-
+        std::string remote_file_url;
+        // The file name of the variant column with the inverted index contains %
+        // such as: 020000000000003f624c4c322c568271060f9b5b274a4a95_0_10133@properties%2Emessage.idx
+        //  {rowset_id}_{seg_num}_{index_id}_{variant_column_name}{%2E}{extracted_column_name}.idx
+        // We need to handle %, otherwise it will cause an HTTP 404 error.
+        if (file_name.find('%') != std::string::npos) {
+            auto escaped_file_name = file_name;
+            boost::replace_all(escaped_file_name, "%", "%25");
```

Comment:
why not use generic url encoding for whole file_nme

Reply:
good idea